lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -60,13 +60,26 @@ struct RvalueExprVisitor {
 
   // Handle named values, such as references to declared variables.
   Value visit(const slang::ast::NamedValueExpression &expr) {
-    if (auto value = context.valueSymbols.lookup(&expr.symbol))
-      return isa<moore::NamedConstantOp>(value.getDefiningOp())
-                 ? value
-                 : builder.create<moore::ReadOp>(
-                       loc,
-                       cast<moore::RefType>(value.getType()).getNestedType(),
-                       value);
+    if (auto value = context.valueSymbols.lookup(&expr.symbol)) {
+      if (auto refType = dyn_cast<moore::RefType>(value.getType()))
+        value =
+            builder.create<moore::ReadOp>(loc, refType.getNestedType(), value);
+      return value;
+    }
+
+    // Try to materialize constant values directly.
```

Comment:
I noticed that you mentioned 
> for example a parameter or an enum variant, and materialize that constant directly as a moore.constant op.

The `Parameter` is still stored in the `ValueSymbol`, then return it(`namedConstant`), rather than its constant value.

Reply:
As if I misunderstand again, that is materializing the constant directly as a `moore.constant`. Sorry that! :sweat_smile:

Reply:
Yeah this is basically a fall-back for when constants are used that have not been lowered into a `moore.named_constant`. But if there is a `named_constant` available, this code will still prefer that :smiley: